api/src/test/java/org/openmrs/module/ModuleFileParserUnitTest.java

```diff
@@ -67,9 +67,7 @@
 
 	private static DocumentBuilder documentBuilder;
 
-	@Rule
-	public ExpectedException expectedException = ExpectedException.none();
-
+	
 	@Rule
 	public TemporaryFolder temporaryFolder = new TemporaryFolder();
 
```

Comment:
please this classs is failling because of the

Reply:
what does it return?

Reply:
does it return the module name at the end of the message?

I believe I adapted
```java
	private void expectModuleExceptionWithMessage(Executable executable, String expectedMessage) {
		ModuleException thrown = assertThrows(ModuleException.class, executable);
		assertThat(thrown.getMessage(), startsWith(expectedMessage));
	}
```

to use `startsWith`
see https://github.com/openmrs/openmrs-core/commit/e20474f2ec7bc39eda9070557e5103ce92b40d75#diff-2070c992acbb365e64864e095104fb74L1203-L1206

Reply:
it does not work, in fact I deleted the function and I call


             ModuleException thrown = assertThrows(ModuleException.class, executable);
		assertThat(thrown.getMessage(), startsWith(expectedMessage)) 

directly in each test.
![Screenshot from 2020-07-02 11-38-29](https://user-images.githubusercontent.com/39851246/86349723-818f8800-bc59-11ea-9bb0-27ab8363fc96.png)

Reply:
The problem is this function :
![Screenshot from 2020-07-02 11-39-13](https://user-images.githubusercontent.com/39851246/86349850-b0a5f980-bc59-11ea-9117-ba84a8f65158.png)

![Screenshot from 2020-07-02 11-40-59](https://user-images.githubusercontent.com/39851246/86349947-d16e4f00-bc59-11ea-972b-94eb160afb69.png)
its does not return the value it must return(messageKey).
thanks @teleivo .

Reply:
this is one of the message : 
![Screenshot from 2020-07-02 11-54-25](https://user-images.githubusercontent.com/39851246/86350618-e13a6300-bc5a-11ea-8d5e-02d5f36fcfc6.png)

Reply:
I am looking into why the mock only returns null 👀 

In any case I would advocate for
```java
	private void expectModuleExceptionWithMessage(Executable executable, String expectedMessage) {
		ModuleException thrown = assertThrows(ModuleException.class, executable);
		assertThat(thrown.getMessage(), startsWith(expectedMessage));
	}
```

instead of changing the messageKey like you did

```java
		String messageKey = "Module.error.cannotParseConfigFile Module: modulewithoutconfig.omod";
		whenGettingMessageFromMessageSourceServiceWithKeyReturnSameKey(messageKey);
``` 


Since if you look at the old ExpectedException the `expectMessage` was only checking for the substring. The following is a successful test

```java	
	@Test
	public void some() {
		expectedException.expect(IllegalArgumentException.class);
		expectedException.expectMessage("foo");
		throw new IllegalArgumentException("foo some");
	}
```

So `assertThat(thrown.getMessage(), startsWith(expectedMessage));` would be along those lines.